core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -26,26 +28,46 @@ public function __construct(ResponseContextAccessor $responseContextAccessor)
         $this->responseContextAccessor = $responseContextAccessor;
     }
 
-    public function createWebpageResponseContext(): WebpageResponseContext
+    public function createResponseContext(): ResponseContext
     {
-        $context = new WebpageResponseContext();
+        $context = new ResponseContext();
+
         $this->responseContextAccessor->setResponseContext($context);
 
         return $context;
     }
 
-    public function createResponseContext(): ResponseContext
+    public function createWebpageResponseContext(): ResponseContext
     {
-        $context = new ResponseContext();
-        $this->responseContextAccessor->setResponseContext($context);
+        $context = $this->createResponseContext();
+        $context->add(new HtmlHeadBag());
 
         return $context;
     }
 
-    public function createContaoWebpageResponseContext(PageModel $pageModel): ContaoWebpageResponseContext
+    public function createContaoWebpageResponseContext(PageModel $pageModel): ResponseContext
     {
-        $context = new ContaoWebpageResponseContext($pageModel);
-        $this->responseContextAccessor->setResponseContext($context);
+        $context = $this->createWebpageResponseContext();
+
+        if (!$context->has(HtmlHeadBag::class)) {
+            return $context;
+        }
```

Comment:
How could this ever be true?

Reply:
Defensive coding. If we were to introduce some kind of event in the future or so, who knows what's gonna happen here.

Reply:
But you cannot remove a bag, can you?